ComponentKit/Utilities/CKComponentAction.h

```diff
@@ -134,6 +142,16 @@ class CKTypedComponentAction : public CKTypedComponentActionBase {
   }
 
   /**
+   Allows demoting an action to a simpler action while supplying defaults for the values that won't be passed in.
+   */
+  template<typename... U, bool separator = false, class = typename std::enable_if<!detail::any_are_reference<U...>::value>::type>
```

Comment:
I'm pretty sure this can be simplified a bit. Also, although enable_if is the right tool generally, I'm in favor of static asserts instead for our obj-c++ codebase since a lot of people aren't C++ gurus here, and with static assert you can provide a really nice compiler error telling them explicitly what to do.